web/app/js/components/Navigation.jsx

```diff
@@ -318,6 +322,17 @@ class NavigationBase extends React.Component {
       .catch(this.handleApiError);
   }
 
+  checkMulticlusterExtension() {
+    this.api.setCurrentRequests([this.api.fetchExtensions()]);
+
+    this.serverPromise = Promise.all(this.api.getCurrentPromises())
+      .then(([extList]) => {
+        const mcExists = _find(extList, e => e.extensionName === multiclusterExtensionName);
+        this.setState({ showGatewayLink: mcExists || false });
+      })
+      .catch(this.handleApiError);
+  }
```

Comment:
This can be simplified then we use the `GetNamespaceWithExtensionLabel` method as explained below.